indexer/services/ender/__tests__/handlers/order-fills/order-handler.test.ts

```diff
@@ -1359,6 +1360,238 @@ describe('OrderHandler', () => {
+      orderFlags: ORDER_FLAG_SHORT_TERM.toString(),
+      timeInForce: IndexerOrder_TimeInForce.TIME_IN_FORCE_IOC,
+      reduceOnly: true,
+      clientMetadata: 0,
+    });
+
+    const fillAmount: number = takerQuantums;
+    const orderFillEvent: OrderFillEventV1 = createOrderFillEvent(
+      makerOrderProto,
+      takerOrderProto,
+      fillAmount,
+      fillAmount,
+      fillAmount,
+    );
+    const kafkaMessage: KafkaMessage = createKafkaMessageFromOrderFillEvent({
+      orderFillEvent,
+      transactionIndex,
+      eventIndex,
+      height: parseInt(defaultHeight, 10),
+      time: defaultTime,
+      txHash: defaultTxHash,
+    });
+
+    await Promise.all([
+      // initial position for subaccount 1
+      PerpetualPositionTable.create({
+        ...defaultPerpetualPosition,
+        perpetualId: testConstants.defaultPerpetualMarket3.id,
+      }),
+      // initial position for subaccount 2
+      PerpetualPositionTable.create({
+        ...defaultPerpetualPosition,
+        subaccountId: testConstants.defaultSubaccountId2,
+        perpetualId: testConstants.defaultPerpetualMarket3.id,
+      }),
+    ]);
+
+    await onMessage(kafkaMessage);
+
+    const makerOrderId: string = OrderTable.orderIdToUuid(makerOrderProto.orderId!);
+    const takerOrderId: string = OrderTable.orderIdToUuid(takerOrderProto.orderId!);
+
+    const [makerOrder, takerOrder]: [
+      OrderFromDatabase | undefined,
+      OrderFromDatabase | undefined
+    ] = await Promise.all([
+      OrderTable.findById(makerOrderId),
+      OrderTable.findById(takerOrderId),
+    ]);
+
+    expect(makerOrder).toBeDefined();
+    expect(takerOrder).toBeDefined();
+
+    // maker order is partially filled
+    expect(makerOrder!.status).toEqual(OrderStatus.CANCELED);
+    // taker order is fully filled
+    expect(takerOrder!.status).toEqual(OrderStatus.FILLED);
+  });
+
+  it.each([
+    [
+      'limit',
+      'via knex',
+      false,
+      IndexerOrder_TimeInForce.TIME_IN_FORCE_UNSPECIFIED,
+    ],
+    [
+      'limit',
+      'via SQL function',
+      true,
+      IndexerOrder_TimeInForce.TIME_IN_FORCE_UNSPECIFIED,
+    ],
+    [
+      'post-only',
+      'via knex',
+      false,
+      IndexerOrder_TimeInForce.TIME_IN_FORCE_POST_ONLY,
+    ],
+    [
+      'post-only',
+      'via SQL function',
+      true,
+      IndexerOrder_TimeInForce.TIME_IN_FORCE_POST_ONLY,
+    ],
+  ])('correctly sets status for short term %s orders (%s)', async (
+    _orderType: string,
+    _name: string,
+    useSqlFunction: boolean,
+    timeInForce: IndexerOrder_TimeInForce,
+  ) => {
+    config.USE_ORDER_HANDLER_SQL_FUNCTION = useSqlFunction;
+    const transactionIndex: number = 0;
+    const eventIndex: number = 0;
+    const makerQuantums: number = 100;
+    const makerSubticks: number = 1_000_000;
+
+    const makerOrderProto: IndexerOrder = createOrder({
+      subaccountId: defaultSubaccountId,
```

Comment:
most of these fields seem to be shared between tests. Can we defined a constant variable and re-use it between tests?

Reply:
I think this refactor goes beyond the scope of this ticket, I'll file a ticket to clean up this file: https://linear.app/dydx/issue/IND-440/clean-up-our-tests